scripts/libpng/1.6.25/script.sh

```diff
@@ -5,20 +5,42 @@ MASON_VERSION=1.6.25
 MASON_LIB_FILE=lib/libpng.a
 MASON_PKGCONFIG_FILE=lib/pkgconfig/libpng.pc
 
+# Used when cross compiling to cortex_a9
+ZLIB_SHARED_VERSION=1.2.8
+
 . ${MASON_DIR}/mason.sh
 
 function mason_load_source {
     mason_download \
-        http://downloads.sourceforge.net/project/libpng/libpng16/${MASON_VERSION}/libpng-${MASON_VERSION}.tar.gz \
+        http://downloads.sourceforge.net/project/libpng/libpng16/older-releases/${MASON_VERSION}/libpng-${MASON_VERSION}.tar.gz \
         a88b710714a8e27e5e5aa52de28076860fc7748c
 
     mason_extract_tar_gz
 
     export MASON_BUILD_PATH=${MASON_ROOT}/.build/libpng-${MASON_VERSION}
 }
 
+function mason_prepare_compile {
+    if [[ ${MASON_PLATFORM_VERSION} == "cortex_a9" ]]; then
+        cd $(dirname ${MASON_ROOT})
+        ${MASON_DIR}/mason install zlib_shared ${ZLIB_SHARED_VERSION}
+        ${MASON_DIR}/mason link zlib_shared ${ZLIB_SHARED_VERSION}
+
+        MASON_ZLIB_CFLAGS="$(${MASON_DIR}/mason cflags zlib_shared ${ZLIB_SHARED_VERSION})"
+        MASON_ZLIB_LDFLAGS="-L$(${MASON_DIR}/mason prefix zlib_shared ${ZLIB_SHARED_VERSION})/lib"
+    fi
```

Comment:
👍  Can you add a brief comment about why this is needed for cortex? (for our future selves when we forget)